parser-typechecker/src/Unison/Util/Pretty.hs

```diff
@@ -448,6 +449,9 @@ num n = fromString (show n)
 string :: IsString s => String -> Pretty s
 string = fromString
 
+endSentence :: IsString s => String -> Pretty s
+endSentence str = fromString (str ++ ".")
```

Comment:
@pete-ts I'd do this with:

```Haskell
endSentence :: IsString s => Pretty s -> Pretty s
endSentence p = group (p <> ".")
```

The `group` here will not get broken up when spaces are inserted by `wrap`.

Reply:
@pchiusano I have a fix with the signature `IsString s => Pretty s -> Pretty s`, but that means that the call to `P.string` has to happen in the do block and it might look a bit "noisier" than `IsString s => String -> Pretty s`, which can get applied without the need to `endSentence . string $ dir`.

Reply:
Understood, and I think that's fine.